### DISCUSSION TITLE
Scrolling with the keyboard.

### DISCUSSION THREAD

**Discussion by AdreamCZ** [bczsalba/pytermgui#87] [Q&A]
Hello,
I have a window with several buttons (added dynamically, so I don't know how many beforehand ) under each other. Overflow is set to SCROLL which works fine when navigating with the mouse, but when I want to select these buttons with the arrow keys and I select a button that is offscreen, the "list" doesn't scroll down and the selected button is not visible. 
How  should I implement that the list scrolls down (or up), when a button that is offscreen is selected ?
Thanks

**Comment by bczsalba** [comment]
Oh golly, I had an answer written up to this days ago!

At the moment there is no (easy) way to detect when a widget goes off screen sadly. At some point in the near-ish future we will have a better scrolling implementation that _will_ allow this to happen, however.

You can see an example of me hacking around this same limitation in the `InputField` source:

https://github.com/bczsalba/pytermgui/blob/9e5e7082480e1893a4d4b765f0f6793c10bfe9a6/pytermgui/widgets/input_field.py#L428-L457

If you want to play around with it, I would suggest testing widgets against the container's `content_dimensions` property. I tried my hand at quickly hacking something together, but I didn't have the time to make it work too well sadly.
